docs/architecture/overall.md

```diff
@@ -0,0 +1,73 @@
+# Overall Architecture
+
+Screwdriver is a collection of services that facilitate the workflow for
+Continuous Delivery pipelines.
+
+![Workflow](assets/workflow.png)
+
+## Workflow
+
+1. **Commit new code**
+
+    User starts a new build by one of the following operations:
+
+    - User pushes code to GitHub on the monitored branch
```

Comment:
Maybe add a part about creating a screwdriver project. What's required, and what it means with github and describe the "monitored branch".

And then in this commit code, reference that